python/graphstorm/data/ogbn_datasets.py

```diff
@@ -176,58 +187,58 @@ def process(self):
+            g.edges[self.edge_type].data['train_mask'] = th.full((int_edges,), True, dtype=th.bool)
             if self.reverse_edge:
-                g.edges["rev-interacts"].data['train_mask'] = th.full((int_edges,), True,
+                g.edges[self.edge_type].data['train_mask'] = th.full((int_edges,), True,
                                                                       dtype=th.bool)
         else:
             # the validation pct is 0.1
             val_pct = 0.1
             train_pct = self.edge_pct - val_pct
             # the test is 1 - the rest
-            g.edges["interacts"].data['train_mask'] = th.full((int_edges,), False, dtype=th.bool)
-            g.edges["interacts"].data['val_mask'] = th.full((int_edges,), False, dtype=th.bool)
-            g.edges["interacts"].data['test_mask'] = th.full((int_edges,), False, dtype=th.bool)
-            g.edges["interacts"].data['train_mask'][: int(int_edges*train_pct)] = True
-            g.edges["interacts"].data['val_mask'][int(int_edges*train_pct):
+            g.edges[self.edge_type].data['train_mask'] = th.full((int_edges,), False, dtype=th.bool)
+            g.edges[self.edge_type].data['val_mask'] = th.full((int_edges,), False, dtype=th.bool)
+            g.edges[self.edge_type].data['test_mask'] = th.full((int_edges,), False, dtype=th.bool)
+            g.edges[self.edge_type].data['train_mask'][: int(int_edges*train_pct)] = True
+            g.edges[self.edge_type].data['val_mask'][int(int_edges*train_pct):
                                                   int(int_edges*self.edge_pct)] = True
-            g.edges["interacts"].data['test_mask'][int(int_edges*self.edge_pct):] = True
+            g.edges[self.edge_type].data['test_mask'][int(int_edges*self.edge_pct):] = True
 
             if self.reverse_edge:
-                g.edges["rev-interacts"].data['train_mask'] = th.full((int_edges,), False,
+                g.edges[self.rev_edge_type].data['train_mask'] = th.full((int_edges,), False,
                                                                       dtype=th.bool)
-                g.edges["rev-interacts"].data['val_mask'] = th.full((int_edges,), False,
+                g.edges[self.rev_edge_type].data['val_mask'] = th.full((int_edges,), False,
                                                                     dtype=th.bool)
-                g.edges["rev-interacts"].data['test_mask'] = th.full((int_edges,), False,
+                g.edges[self.rev_edge_type].data['test_mask'] = th.full((int_edges,), False,
                                                                      dtype=th.bool)
-                g.edges["rev-interacts"].data['train_mask'][: int(int_edges * train_pct)] = True
-                g.edges["rev-interacts"].data['val_mask'][int(int_edges*train_pct):
+                g.edges[self.rev_edge_type].data['train_mask'][: int(int_edges * train_pct)] = True
```

Comment:
This is not correct when the graph is homogeneous. When it is a homogeneous graph,`g.edges[self.edge_type]` and `g.edges[self.rev_edge_type]` are identical . So you are setting the same set of edges to True.
Can you check whether `(self.node_type, self.rev_edge_type, self.node_type): (dst, src)` is appended at the end of `(self.node_type, self.edge_type, self.node_type): (src, dst)`.

Reply:
The logic here should be right. I have already added the `(self.node_type, self.rev_edge_type, self.node_type): (dst, src)` when adding reverse edge option. So even if they are set the same set of edges to True, it should be fine, we expect to do so. I add a if condition here to not generate the mask again when it is homogeneous graph. I think then it would be fine.